src/components/input/TextComponent.tsx

```diff
@@ -146,30 +160,37 @@ const TextComponent: React.FC<TextComponentPropsWithType> = ({
           style={
             {
               WebkitTextSecurity:
-                type == Types.TEXTFIELD && masked && hideText && 'disc',
+                type == Types.TEXTFIELD &&
+                (isMasked || (masked && hideText)) &&
+                'disc',
             } as React.CSSProperties
           }
           disabled={disabled}
           {...rest}
         />
-        {iconElement && type == Types.TEXTFIELD
-        // Clone the iconElement in order to attach className 'tk-input__icon'
-          ? React.cloneElement(iconElement, {
-            className: classNames(
-              'tk-input__icon',
-              iconElement.props.className
-            ),
-          })
+
+        {decoratorElement && type == Types.TEXTFIELD
```

Comment:
I suppose that we consider that anything that comes after the input is the right decorator?